### PR TITLE
adding 'targetNamespace' and 'installMode' fields to addon_type model

### PR DESCRIPTION
**Pull request by zvikorn**
Signed-off-by: Tzvika Avni <tavni@dhcp-2-174.tlv.redhat.com>

### CODE REVIEW

File: model/clusters_mgmt/v1/add_on_type.model [hunk 1]
```diff
@@ -36,4 +36,10 @@ class AddOn {
 
 	// Used to determine how many units of quota an add-on consumes per resource name.
 	ResourceCost Float
+
+	// The namespace name in which the addon is installed.
+    TargetNamespace String
```
Comment (jhernand) [inline model/clusters_mgmt/v1/add_on_type.model]: Why is this needed?

File: model/clusters_mgmt/v1/add_on_type.model [hunk 2]
```diff
@@ -36,4 +36,10 @@ class AddOn {
...
+
+    // The mode in which the addon is installed.
+    InstallMode String
```
Comment (jhernand) [inline model/clusters_mgmt/v1/add_on_type.model]: What are the possible values of this install mode and what do they mean? Consider adding a enumerated type for that:

```
enum AddOnInstallMode {
        // This mode means blah, blah, blah ...
        FirstMode

        // This mode means ....
        SecondMode
}
```

That should go in a separate `add_on_install_mode_type.model` file.

Comment (jhernand) [reply model/clusters_mgmt/v1/add_on_type.model]: Also please remember to indent the code only with tabs.

**Comment by jhernand** [comment]
@vkareh please review.

File: model/clusters_mgmt/v1/add_on_type.model (hunk 1 above)
Comment (zvikorn) [reply model/clusters_mgmt/v1/add_on_type.model]: In the documentation comments, you mean in the .model file itself?
This field is required sine we want to present the consumer with the information of which namespace the addon is installed in. We want to show this the targetNamespace on the addon tab.

Comment (jhernand) [reply model/clusters_mgmt/v1/add_on_type.model]: Yes, I mean in the documentation comments in the .model file.

File: model/clusters_mgmt/v1/add_on_type.model (hunk 2 above)
Comment (vkareh) [reply model/clusters_mgmt/v1/add_on_type.model]: This should be an enum. Currently accepted values should be `AllNamespaces` and `SingleNamespace`.

**Comment by zvikorn** [comment]
@vkareh could you please see if the description in each model is correct?

**Comment by vkareh** [comment]
LGTM! thanks!
